packages/bot-web-ui/src/pages/dashboard/announcements/announcements.tsx

```diff
@@ -133,6 +144,9 @@ const Announcements = observer(({ is_mobile, is_tablet, handleTabChange }: TAnno
         if (selected_announcement?.should_toggle_modal) {
             toggleLoadModal();
         }
+        if (selected_announcement?.should_handle_primary_action) {
+            handlePrimaryAction();
+        }
```

Comment:
If we have to switch tab we can utilize the `switch_tab_on_confirm` config value and to toggle the modal we can use the `should_toggle_modal`. The rest of the additional logic specific to the announcement (for example the following code) should be inside the  `onConfirm` property of the config.
`updateLocalStorage('UPDATED_QUICK_STRATEGY_MODAL_ANNOUNCE');
rudderStackSendAnnouncementClickEvent({ announcement_name: localize('Updated: Quick Strategy Modal') });`

Reply:
done ✅